manager/orchestrator/restart/restart.go

```diff
@@ -190,6 +197,83 @@ func (r *Supervisor) Restart(ctx context.Context, tx store.Tx, cluster *api.Clus
 	return nil
 }
 
+// TaskRestartDelay calculates and returns the next delay based t's RestartPolicy.
+// It also returns whether the caller should randomized the value.
+// Important: It is the caller's responsibility to randomize the returned duration.
+func (r *Supervisor) TaskRestartDelay(ctx context.Context, t *api.Task) (time.Duration, bool, error) {
```

Comment:
I'm curious why you decided to put the decision to randomize outside the function ?

It seems cleaner to keep it inside this function and get rid of the bool return value.

Reply:
There were concerns regarding how to test this function without having to use statistical analysis to get reliable results, and this was suggested as one of the more straightforward options.